File: src/main/java/ffblockly/meta/FFBlockly.java
package ffblockly.meta;


import java.nio.ByteBuffer;
import java.nio.charset.StandardCharsets;
import java.util.Map;
import java.util.Set;
import java.util.TreeMap;

import static java.lang.Math.abs;

/**
 */
public interface FFBlockly {
    Map<String, String> typesMap = new TreeMap<String, String>() {{
        put("string", "String");
        put("int", "Number");
        put("boolean", "Boolean");
        put("float", "Number");
        put("double", "Number");
        put("int64", "Number");
        put("int32", "Number");
        put("short", "Number");
        put("byte", "Number");
    }};

    static int colorForString(String filterName) {
        return  filterName.hashCode()% 360 ;
    }


    Set<FFBlock> getModel();
}

File: src/main/java/ffblockly/meta/FFBlockly.kt
package ffblockly.meta

import java.util.*

/**
 */
interface FFBlockly {
    val model: Set<FFBlock?>?

    companion object {
        @JvmStatic
        fun colorForString(filterName: String): Int {
            return filterName.hashCode() % 360
        }

        @JvmField
        val typesMap: SortedMap<String, String> =
        sortedMapOf(
                ("string" to "String")
         ,  ("int" to "Number")
         ,  ("boolean" to "Boolean")
         ,  ("float" to "Number")
         ,  ("double" to "Number")
         ,  ("int64" to "Number")
         ,  ("int32" to "Number")
         ,  ("short" to "Number")
         ,  ("byte" to "Number")
        )
    }

}